Assets/MRTK.ThirdParty/MRTK-Quest/Editor/MRTKQuestConfigurationChecker.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace prvncher.MixedReality.Toolkit.OculusQuestInput
+{
+    /// <summary>
+    /// Class that checks if the Oculus Integration Assets are present and configures the project if they are.
+    /// </summary>
+    [InitializeOnLoad]
+    static class OculusConfigurationChecker
+    {
+        // The presence of the OculusProjectConfig.asset is used to determine if the Oculus Integration Assets are in the project.
+        private const string OculusIntegrationProjectConfig = "OculusProjectConfig.asset";
+        private static readonly string[] Definitions = { "OCULUSINTEGRATION_PRESENT" };
+
+        static OculusConfigurationChecker()
```

Comment:
Per the issue here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/8188

We should try to move this to a manual step because this adds overhead to compile/play mode entering.

Basically we should do what was done here:

https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8287

Note that you might read this and be like "but dude you reverted that change" - yes, that happened because we're using an older version of Unity 2018 on some CI things where ARFoundation stuff causes things to throw up, but shouldn't affect the stuff here with Oculus Quest support.